core/savings.go

```diff
@@ -5,82 +5,130 @@ import (
 	"time"
 
 	"github.com/benbjohnson/clock"
+	"github.com/evcc-io/evcc/tariff"
 	"github.com/evcc-io/evcc/util"
 )
 
+// publisher gives access to the site's publish function
+type publisher interface {
+	publish(key string, val interface{})
+}
+
 // Site is the main configuration container. A site can host multiple loadpoints.
 type Savings struct {
-	log *util.Logger
-
+	log                    *util.Logger
+	clock                  clock.Clock
+	tariffs                tariff.Tariffs
 	started                time.Time // Boot time
 	updated                time.Time // Time of last charged value update
-	chargedTotal           float64   // Energy charged since startup (kWh)
+	chargedGrid            float64   // Grid energy charged since startup (kWh)
 	chargedSelfConsumption float64   // Self-produced energy charged since startup (kWh)
-	Clock                  clock.Clock
+	costGrid               float64   // Cost of charged grid energy (e.g. EUR)
```

Comment:
gridCost
    selfCost

Kommentar vllt. "running total of...."

Reply:
done